File: UniBooks_Backend/UniBooks_Backend/Models/ShoppingCart.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UniBooks_Backend.Models
{
    public class ShoppingCart
    {
        [Key]
        public int ShoppingCart_ID { get; set; }

        [Column(TypeName = "decimal(18, 2)")]
        public decimal Discount { get; set; }

        public int Count { get; set; }

        public decimal TotalAmount { get; set; }

        public int StudentID { get; set; }
        public Student Student { get; set; }

        [Required]
        [Column(TypeName = "decimal(18, 2)")]
        public decimal SubTotal { get; set; }

        // Navigation property: link to shopping cart book
        public List<ShoppingCart_Book> ShoppingCartBook { get; set; }

        // Navigation property: link to shopping cart equipment
        public List<ShoppingCart_Equipment> ShoppingCartEquipment { get; set; }
    }
}

File: Ionic_Student/src/app/shared/shoppingcart.ts
import { OrderLine } from "./order";
import { Books, Book } from "./books";

export class ShoppingCart {
  $id: string = "";
  shoppingCart: {
    $id: string;
    shoppingCart_ID: number;
    subTotal: number;
    count: number;
    totalAmount: number;
    studentId: number;
    discount: number;
    shoppingCartBook: ShoppingCartBook[];
    shoppingCartEquipment: ShoppingCartEquipment[];
    orderLine: OrderLine | null;
  } = {
      $id: "",
      shoppingCart_ID: 0,
      subTotal: 0,
      count: 0,
      studentId: 0,
      totalAmount: 0,
      discount: 0,
      shoppingCartBook: [],
      shoppingCartEquipment: [],
      orderLine: null,
    };
  subTotal: number = 0;
  count: number = 0;
  discount: number = 0;
  totalAmount: number = 0;
  shoppingCart_ID: number = 0;
}


export class BookItems {
  $id: string = "";
  $values: ShoppingCartBook[] = [];
}

export class ShoppingCartBook {
  $id: string = "";
  book_ID: number = 0;
  shoppingCart_ID: number = 0;
  title: string = "";
  image: string = "";
  price: number = 0;
  quantity: number = 0;
}


export class EquipmentItems {
  $id: string = "";
  $values: ShoppingCartEquipment[] = [];
}

export class ShoppingCartEquipment {
  $id: string = "";
  shoppingCart_ID: number = 0;
  equipment_ID: number = 0;
  image: string = "";
  price: number = 0;
  name: string = "";
  quantity: number = 0;
}
